core/src/term/mod.rs

```diff
@@ -1453,6 +1453,19 @@ impl RichTerm {
             truncated
         }
     }
+
+    pub fn find_pos(&self, pos: RawPos) -> Option<RichTerm> {
+        let mut ret = None;
+        self.traverse_ref(&mut |rt: &RichTerm| {
+            if rt.pos.contains(pos) {
+                ret = Some(rt.clone());
+                TraverseControl::<()>::Continue
+            } else {
+                TraverseControl::SkipBranch
+            }
+        });
+        ret
+    }
```

Comment:
I'm wondering if it would be reasonable to construct an interval tree like index structure, so we don't have to traverse the entire term each time. But maybe I'm reinventing the linearization concept now 😅

In any case, we shouldn't do that in this PR; just a thought.

Reply:
I think in this case the linearization can also be linear. The parsed expression nests like `((({} & {}) & {}) & {})` and so the ranges of the various Op2 terms will be like [0, 5], [0, 8], [0, 11], and so on, and only after all of the [0, n] ranges do we start getting ranges like [m, m+2] for the record terms. The linearizer does a binary search for the start of the range and then it does a linear scan among all ranges with that same start. So if you query a range near the start of the file, I think it will be linear.

Anyway, I think it shouldn't be hard to add an interval map for constant-time lookups. I even have a [crate](https://crates.io/crates/range-map) for it.

Reply:
Ok, in the very specific case of querying a range between 0 and the next well-defined item, that's probably true. Still, most of the time, you query a well-defined sub-node, and in that case the current implementation should perform well (for example any of the record operands), while this one always start from the root and navigate through the tree - it doesn't have the possibility of jumping in the middle of the tree, I believe. So querying the `n-th` operator is `O(size of the AST)`. It's also probably possible to do a second binary search as well, to find the closest match, if that's required.

Anyway, as you say, that's something that can be improved later on :+1: